### DISCUSSION TITLE
Weird thing in block rule.! Dev should chage it.

### DISCUSSION THREAD

**Discussion by lessload** [DNSCrypt/dnscrypt-proxy#1723] [Potential issues]
1. I'm using block list from popular site like https://github.com/StevenBlack/hosts but I'm not add `=` on all site. ( =url...)
I found some site was block with no reason, for example "uptime.seby.io" was blocked but there no "uptime.seby.io" or "seby.io" in block list. in query log show as reject. i try to add `=` on all url. The bug was gone. But it still not common sense. Why `*.hostname` equal to `hostname` this is not common sense. You should change it, `*.hostname` should not equal to `hostname`. This really crazy when i update block list and i need to add `=` on all host, every time i update block.

2. Cloak rule was overwrite by block rule?? (from test) it not common sense again. Why people need block it if url store in cloak rule.When you use large host block it will break cloak rule somedays. Cloak rule should overwrite block rule? this more make sense.

**Comment by ghost** [comment]
1. DNSCrypt-Proxy does cname uncloaking, i.e. it can see that "uptime.seby.io" is the same as "[something].compute.amazonaws.com" and the latter is actually blocked in your host list.

**Reply by lessload** [reply]
oh. Great to know that. thank you.
in my opinion It so complex to use or understand Why developer make `*.hostname` equal to `hostname`. Never found something like this in other software.
